### ISSUE TITLE
Export of inconsistent internal headers

### ISSUE THREAD

**Issue by drabmart** [hpc/mpifileutils#549]
The header file mfu_flist_internal.h gets installed (distributed) with the compiled library. (Is this intentional? Why is it called "internal" then?) Anyway, unfortunatelly it is being included by the utils/tools/h5dwalk/h5dwalk.c file (from the HDF5 1.14 project). However, mfu_flist_internal.h unconditionally includes strmap.h, which is not being installed. And that is an error. Either both or none of them should be installed (distributed). If the functions/structures in the mfu_flist_internal.h are supposed to be used externally (like apparently they are in HDF5), then strmap.h has to be installed (distributed) too.

**Comment by adammoody** [comment]
Yes, good point.  Thanks for the report, @drabmart .  The original intent was not to distribute internal headers, but then we did so since it made it easier for others to develop new tools using the ``libmfu`` library.  I think including ``strmap.h`` was then overlooked.  I'll take a look.
